### PR TITLE
Optimize React-in-Vue of immediate react children.

### PR DESCRIPTION
**Pull request by phillbaker**
This is a proof of concept for handling a special case of React components as children of React components in Vue. It works by creating a registry of React components and then checking to see if components are parent/children. 

[Vue's built-in resolution](https://github.com/vuejs/vue/blob/05299610ea3e89ddbcfe4d8ede0c298223766423/src/core/vdom/create-element.js#L105) with [name checking](https://github.com/vuejs/vue/blob/19552a82a636910f4595937141557305ab5d434e/dist/vue.runtime.common.js#L1491) isn't available as a utility, so included similar code.

Ref: https://github.com/akxcv/vuera/issues/19.

### CODE REVIEW

**Comment by phillbaker** [comment]
@akxcv any thoughts here?

**Comment by akxcv** [comment]
Hi! I skimmed through your code a few times, and, frankly, it's quite hard to understand what is going on. Could you please lay out the general strategy in plain English?

I think it should be possible to detect whether the component is from Vue or from React in the wrappers themselves (right now they are just blindly wrapping whatever we pass). Correct me if I'm wrong, but it is much simpler than using a registry of some sort.

**Comment by phillbaker** [comment]
@akxcv thanks for taking a look.

> I think it should be possible to detect whether the component is from Vue or from React in the wrappers themselves (right now they are just blindly wrapping whatever we pass). Correct me if I'm wrong, but it is much simpler than using a registry of some sort.

I initially thought so and https://github.com/akxcv/vuera/pull/44 may help here, but I wasn't able to do this in the wrapper objects themselves because all components look the same - it seems only possible to determine the true type of a component in the plugins. E.g. in Vue, all react components are converted to vue components in `VuePlugin`.

> I skimmed through your code a few times, and, frankly, it's quite hard to understand what is going on. Could you please lay out the general strategy in plain English?

* in VuePlugin, maintain a registry of which components are actually react components (under the hood)
* in React.js look up the component in the registry by any of the various names it could be referenced
  * if the component is a react component and all of its direct children are react components use react directly to render the children
* update Vue.js to use unique component names to avoid Vue reusing instances (frankly I don't understand the Vue internals enough to understand why this is necessary)

This was definitely a proof of concept and happy to refactor / take a different direction!

**Comment by akxcv** [comment]
Ah, the problem is that the Vue plugin is detecting a React component and instantly wrapping it into a React wrapper, right?
Again, sorry for taking a long time...

**Comment by phillbaker** [comment]
That's correct!

I briefly tried to set additional properties on the created Vue component
to be able to detect it later, but that didn't end up working, I forget
exactly why.

**Comment by akxcv** [comment]
Then, I think, we need to rethink the whole flow of Vuera to be more robust, because the current algorithm is kind of hacky. I'll try to think about it soon.
